src/emc/usr_intf/axis/scripts/axis.py

```diff
@@ -1304,15 +1304,25 @@ def open_file_guts(f, filtered=False, addrecent=True):
 
 def activate_ja_widget(i, force=0):
     if not force and not manual_ok(): return
-    if get_jog_mode():
-        # free jogging (joints) (only accept integers here)
+    if get_jog_mode() and trivkinstype != "single":
+        # free jogging (joints) if:
+        #   trivkins config and kinstype = both
+        #   non-trivkins config
+        # only accept integers here
         if type(i) != type(0): return
         if i >= num_joints: return
         widget = getattr(widgets, "joint_%d" % i)
     else:
-        # teleop jogging (axes) (letters are special case for key bindings)
-        if type(i) == type(""): letter = i
```

Comment:
Thanks, I had not heard of isinstance().
Considering I do need to make changes would you prefer that I also change this at the same time?